### PR TITLE
Using the batch size from pelias-config.

### PR DESCRIPTION
**Pull request by mansoor-sajjad**
:wave: 
Please read the details on PR in [pelias-config#139](https://github.com/pelias/config/pull/139).
This PR is dependent on [pelias-config#139](https://github.com/pelias/config/pull/139).

This PR needs to be updated with the latest version of `pelias-config`, after the above referenced PR get merged.

---
#### Here's the reason for this change :rocket:
Hardcoded batchSize of 500 is not optimal in all circumstances.

---
#### Here's what actually got changed :clap:
We are using the newly added `batchSize` configuration option from `pelias-config`.
Updated the existing and added the new tests.

### CODE REVIEW

File: test/Batch.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use strict';
```
Comment (orangejulius) [inline test/Batch.js:1]: Thanks for splitting these tests out into a new file, good choice 👍 

Minor details: `'use strict';` is no longer needed, as I recall it's a relic from the Node.js 6 and earlier days. Goes to show what Peter said about this code not having been touched much in a while.

**Comment by mansoor-sajjad** [comment]
Hei @missinglink,

Thanks for the review.
I have implemented the suggested improvements and fixes.
Can you please have a look at new changes?

**Review by orangejulius — APPROVED** [review]
Looks good overall, all of @missinglink's suggestions were sound.

One thing we need to make sure to do before this is merged is make sure there is a proper semantic release commit so a new version is released. @missinglink how do we want to do that?

**Comment by mansoor-sajjad** [comment]
@missinglink
Do you have the opportunity to look into the 'semantic release commit', as mentioned by @orangejulius ?

**Comment by mansoor-sajjad** [comment]
@missinglink
Do you have the opportunity to look into the 'semantic release commit', as mentioned by @orangejulius ?

**Comment by missinglink** [comment]
Hi @mansoor-sajjad, can we avoid passing `peliasConfig` into `Batch()`?

```diff
diff --git a/package.json b/package.json
index 370c252..9d1d8da 100644
--- a/package.json
+++ b/package.json
@@ -38,7 +38,7 @@
   "dependencies": {
     "@hapi/joi": "^16.0.0",
     "elasticsearch": "^16.0.0",
-    "pelias-config": "^4.5.0",
+    "pelias-config": "^5.2.0",
     "pelias-logger": "^1.2.1",
     "through2": "^3.0.0"
   },
diff --git a/src/Batch.js b/src/Batch.js
index 9ea8fbb..fe89eb4 100644
--- a/src/Batch.js
+++ b/src/Batch.js
@@ -1,8 +1,13 @@
 const Task = require('./Task');
 const _ = require('lodash');
+const config = require('pelias-config');
 
-function Batch(opts, peliasConfig) {
-  this._size = _.defaults({}, opts, peliasConfig.dbclient).batchSize;
+function Batch(opts) {
+  this._size = (
+    _.get(opts, 'batchSize') ||
+    _.get(config.generate(), 'dbclient.batchSize') ||
+    500 // maximum records per batch
+  );
   this._slots = [];
   this.retries = 0;
   this.status = 999;
@@ -18,4 +23,4 @@ Batch.prototype.push = function( record ){
   this._slots.push( new Task( record ) );
 };
 
-module.exports = Batch;
\ No newline at end of file
+module.exports = Batch;
diff --git a/src/BatchManager.js b/src/BatchManager.js
index c36fc43..5055fb3 100644
--- a/src/BatchManager.js
+++ b/src/BatchManager.js
@@ -1,7 +1,6 @@
 const Batch = require('./Batch');
 const transaction = require('./transaction');
 const pelias_logger = require('pelias-logger');
-const peliasConfig = require('pelias-config').generate();
 
 const Stats = require('./stats');
 
@@ -20,7 +19,7 @@ function BatchManager( opts ){
   this._stats = new Stats(this._logger);
 
   // internal variables
-  this._current = new Batch( this._opts, peliasConfig );
+  this._current = new Batch( this._opts );
   this._transient = 0;
   this._resumeFunc = undefined;
 
@@ -98,7 +97,7 @@ BatchManager.prototype._dispatch = function( batch, next ){
 
 BatchManager.prototype.flush = function(next){
   this._dispatch( this._current, next );
-  this._current = new Batch( this._opts, peliasConfig );
+  this._current = new Batch( this._opts );
 };
 
 // call this on stream end
diff --git a/test/Batch.js b/test/Batch.js
index 75287bd..3c67ad1 100644
--- a/test/Batch.js
+++ b/test/Batch.js
@@ -13,7 +13,7 @@ module.exports.tests.validate = function (test) {
       batchSize: 200
     };
 
-    const batch = new Batch(opts, peliasConfig.generate());
+    const batch = new Batch(opts);
 
     t.equals(batch.free(), 200, 'opts should be prioritised over config');
     t.end();
@@ -25,7 +25,7 @@ module.exports.tests.validate = function (test) {
 
     process.env.PELIAS_CONFIG = path.resolve(__dirname + '/test-config.json');
 
-    const batch = new Batch(undefined, peliasConfig.generate());
+    const batch = new Batch();
 
     t.equals(batch.free(), 1000, 'batch size from config should be used when opts do not have it');
     t.end();
@@ -35,7 +35,7 @@ module.exports.tests.validate = function (test) {
 
   test('default batch size from config should be used', function (t) {
 
-    const batch = new Batch(undefined, peliasConfig.generate());
+    const batch = new Batch();
 
     t.equals(batch.free(), 500, 'default batch size should be used when no batch size is configured');
     t.end();
```

**Comment by missinglink** [comment]
It's a pity we need to call `generate()` on every instantiation of `Batch()`, but otherwise it's impossible to test properly

**Comment by missinglink** [comment]
Hi @mansoor-sajjad how about this version which avoids calling `generate()` through the introduction of static state in a new module called `config.js`?

https://github.com/pelias/dbclient/pull/126

Can you please have a look and check that it does what you need?

**Comment by mansoor-sajjad** [comment]
Hi @missinglink,

Thank for implementing #126.
Yes it fulfills our need.

**Pull request closed without merging**